File: packages/rest/functions/ruby/hello.rb
load File.join(__dir__, 'vendor/bundle/bundler/setup.rb')

require 'json'

def handler(event:, context:)
  puts "event #{JSON.pretty_generate(event)}"
  email = event['requestContext']['authorizer']['email']
  if email.nil?
    body_json = { 'hello': 'Hej! from ruby' }
  else
    body_json = { 'hello': "Hej #{email}! from ruby" }
  end
  {
    statusCode: 200,
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.generate(body_json)
  }
end

File: packages/rest/functions/python/hello.py
# -*- coding: utf-8 -*-

import json
import os
import traceback

from pprint import pprint


def handler(event, context):
    pprint({"event": event})
    email = None
    try:
        email = event["requestContext"]["authorizer"]["email"]
    except KeyError:
        pass
    if email:
        body_json = {"hello": f"Hej {email}! from python"}
    else:
        body_json = {"hello": "Hej! from python"}

    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps(body_json),
    }
